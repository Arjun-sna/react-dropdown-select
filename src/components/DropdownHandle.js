import React from 'react';
import styled from '@emotion/styled';

const dropdownHandle = ({ parentProps, parentState, parentMethods }) => parentProps.dropdownHandleRenderer ? (
  parentProps.dropdownHandleRenderer(parentProps, parentState, parentMethods)
) : (
  <DropdownHandleComponent
    tabIndex="-1"
    onClick={() => parentMethods.dropDown()}
    onKeyPress={() => parentMethods.dropDown()}
    onKeyDown={() => parentMethods.dropDown()}
    className="react-dropdown-select-dropdown-handle">
    {parentState.dropdown ? (
      <React.Fragment>&rsaquo;</React.Fragment>
    ) : (
      <React.Fragment>&lsaquo;</React.Fragment>
    )}
  </DropdownHandleComponent>
);

dropdownHandle.propTypes = {};

const DropdownHandleComponent = styled.div`
  width: 20px;
  text-align: center;
  line-height: 25px;
  transform: rotate(-90deg);
  cursor: pointer;
  font-size: 26px;
  :focus {
    outline: none;
    color: deepskyblue;
  }
`;

export default dropdownHandle;
