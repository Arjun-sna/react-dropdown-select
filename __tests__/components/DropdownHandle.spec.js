import React from 'react';
import renderer from 'react-test-renderer';

import DropdownHandle from '../../src/components/DropdownHandle';

const props = {
  parentProps: {
    dropdownHandleRenderer: null
  },
  parentState: {
    dropdown: false
  }
};

it('<DropdownHandle/> renders correctly', () => {
  const tree = renderer.create(<DropdownHandle {...props}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
