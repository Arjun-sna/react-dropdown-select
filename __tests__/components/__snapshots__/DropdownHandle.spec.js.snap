// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownHandle/> renders correctly 1`] = `
<div
  className="react-dropdown-select-dropdown-handle css-2zkru-DropdownHandleComponent e1vudypg0"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  tabIndex="-1"
>
  <svg
    fill="currentColor"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 40 40"
  >
    <g>
      <path
        d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z"
      />
    </g>
  </svg>
</div>
`;
